payments-core/src/main/java/com/stripe/android/networking/StripeApiRepository.kt

```diff
@@ -177,24 +177,28 @@ class StripeApiRepository @JvmOverloads internal constructor(
         clientSecret: String,
         options: ApiRequest.Options,
         expandFields: List<String>
-    ): StripeIntent {
+    ): Result<StripeIntent> {
         return when {
             PaymentIntent.ClientSecret.isMatch(clientSecret) -> {
-                requireNotNull(
-                    retrievePaymentIntent(clientSecret, options, expandFields)
-                ) {
-                    "Could not retrieve PaymentIntent."
+                runCatching {
+                    requireNotNull(
```

Comment:
This will be updated in a follow-up pull request where `retrievePaymentIntent` and `retrieveSetupIntent` will be made to return `Result<T>`.